local-cli/runIOS/RunIOSInternals.js

```diff
@@ -0,0 +1,272 @@
+    try {
+      child_process.spawnSync('xcrun', ['instruments', '-w', selectedSimulator.udid]);
+    } catch (e) {
+      // instruments always fail with 255 because it expects more arguments,
+      // but we want it to only launch the simulator
+    }
+    resolve(selectedSimulator.udid)
+  })
+  .then((udid) => this.buildProject(xcodeProject, udid, scheme, args.configuration, args.packager))
+  .then((appName) => {
+    if (!appName) {
+      appName = scheme;
+    }
+    let appPath = this.getBuildPath(args.configuration, appName);
+    console.log(`Installing ${appPath}`);
+    child_process.spawnSync('xcrun', ['simctl', 'install', 'booted', appPath], {stdio: 'inherit'});
+
+    const bundleID = child_process.execFileSync(
+      '/usr/libexec/PlistBuddy',
+      ['-c', 'Print:CFBundleIdentifier', path.join(appPath, 'Info.plist')],
+      {encoding: 'utf8'}
+    ).trim();
+
+    console.log(`Launching ${bundleID}`);
+    child_process.spawnSync('xcrun', ['simctl', 'launch', 'booted', bundleID], {stdio: 'inherit'});
+  })
+};
+
+RunIOSInternals.prototype.runOnDevice = function(selectedDevice, scheme, xcodeProject, configuration, launchPackager) {
+  return this.buildProject(xcodeProject, selectedDevice.udid, scheme, configuration, launchPackager)
+  .then((appName) => {
+    if (!appName) {
+      appName = scheme;
+    }
+    const iosDeployInstallArgs = [
+      '--bundle', this.getBuildPath(configuration, appName, true),
+      '--id' , selectedDevice.udid,
+      '--justlaunch'
+    ];
+    console.log(`installing and launching your app on ${selectedDevice.name}...`);
+    const iosDeployOutput = child_process.spawnSync('ios-deploy', iosDeployInstallArgs, {encoding: 'utf8'});
+    if (iosDeployOutput.error) {
+      console.log('');
+      console.log('** INSTALLATION FAILED **');
+      console.log('Make sure you have ios-deploy installed globally.');
+      console.log('(e.g "npm install -g ios-deploy")');
+    } else {
+      console.log('** INSTALLATION SUCCEEDED **');
+    }
+  });
+};
+
+RunIOSInternals.prototype.buildProject = function(xcodeProject, udid, scheme, configuration = 'Debug', launchPackager = false) {
+  return new Promise((resolve,reject) =>
+  {
+     var xcodebuildArgs = [
+      xcodeProject.isWorkspace ? '-workspace' : '-project', xcodeProject.name,
+      '-configuration', configuration,
+      '-scheme', scheme,
+      '-destination', `id=${udid}`,
+      '-derivedDataPath', 'build',
+    ];
+    console.log(`Building using "xcodebuild ${xcodebuildArgs.join(' ')}"`);
+    const buildProcess = child_process.spawn('xcodebuild', xcodebuildArgs, this.getProcessOptions(launchPackager));
+    let buildOutput = "";
+    buildProcess.stdout.on('data', function(data) {
+      console.log(data.toString());
+      buildOutput += data.toString();
+    });
+    buildProcess.stderr.on('data', function(data) {
+      console.error(data.toString());
+    });
+    buildProcess.on('close', function(code) {
```

Comment:
nit: use `arrow` functions where possible over functions like that (also applies to other places in the code)